### PR TITLE
MULE-16669: Mule throws error Illegal group reference Exception when …

### PR DESCRIPTION
**Pull request by JuLuzzi**
…trying to pass $ into URI Parameter in HTTP request

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java [hunk 1]
```diff
@@ -6,17 +6,21 @@
...
 import org.mule.module.http.internal.ParameterMap;
 
 import java.util.List;
+import java.util.regex.Pattern;
 
 public class HttpRequesterRequestBuilder extends HttpMessageBuilder
 {
 
-    public ParameterMap getQueryParams(MuleEvent event)
+    ParameterMap getQueryParams(MuleEvent event)
```
Comment (afelisatti) [inline modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java]: This breaks compatibility.

File: modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java [hunk 2]
```diff
@@ -26,7 +30,7 @@ public ParameterMap getHeaders(MuleEvent event)
         return resolveParams(event, HttpParamType.HEADER);
     }
 
-    public String replaceUriParams(String path, MuleEvent event)
+    String replaceUriParams(String path, MuleEvent event)
```
Comment (afelisatti) [inline modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java]: same

File: modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java [hunk 3]
```diff
@@ -39,12 +43,16 @@ public String replaceUriParams(String path, MuleEvent event)
 
             if (uriParamValue == null)
             {
-                throw new NullPointerException(String.format("Expression {%s} evaluated to null.", uriParamName));
+                throw new NullPointerException(format("Expression {%s} evaluated to null.", uriParamName));
             }
 
-            path = path.replaceAll(String.format("\\{%s\\}", uriParamName), uriParamValue);
+            String value = quoteReplacement(uriParamValue);
```
Comment (afelisatti) [inline modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java]: Why was this necessary?

Comment (JuLuzzi) [reply modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java]: It's necessary to solve the dollar sign problem, replaceAll method documentation: "Note that backslashes and dollar signs in the replacement string may cause the results to be different than if it were being treated as a literal replacement string. Dollar signs may be treated as references to captured subsequences as described above, and backslashes are used to escape literal characters in the replacement string."

Comment (afelisatti) [reply modules/http/src/main/java/org/mule/module/http/internal/request/HttpRequesterRequestBuilder.java]: inline it
